src/main/java/com/faforever/client/chat/ChatUserItemController.java

```diff
@@ -224,39 +229,35 @@ private void setAvatarUrl(@Nullable String avatarUrl) {
 
   private void setClanTag(String newValue) {
     // code to test can be inserted here set new Value to some existing clanTag and the players clan...player.setClan("TAG")
-   
+
     if (StringUtils.isEmpty(newValue)) {
       clanMenu.setVisible(false);
       return;
     }
 
 
-      Runnable setMenuItems = new Runnable() {
-        @Override
-        public void run() {
-          clan = clanService.getClanByTag(player.getClan());
-          if (clan == null) {
-            return;
-          }
-
-            MenuItem page = new MenuItem(i18n.get("clan.visitPage"));
-            page.setOnAction(event -> {
-              platformService.showDocument(baseClanWebsite + clan.getClanId());
-              // TODO: Could be viewed in clan section (if implemented)
-            });
+    Runnable setMenuItems = () -> {
+      clan = clanService.getClanByTag(player.getClan());
+      if (clan == null) {
+        return;
+      }
 
-          if (!playerService.isOnline(clan.getLeaderName())) {
-            clanMenu.getItems().addAll(FXCollections.observableArrayList(page));
-            return;
-          }
+      MenuItem page = new MenuItem(i18n.get("clan.visitPage"));
+      page.setOnAction(event -> {
+        platformService.showDocument(baseClanWebsite + clan.getClanId());
+        // TODO: Could be viewed in clan section (if implemented)
+      });
 
-              MenuItem toLeader = new MenuItem(i18n.get("clan.toLeader"));
-          toLeader.setOnAction(event -> onClanTagClicked());
+      if (!playerService.isOnline(clan.getLeaderName())) {
+        clanMenu.getItems().addAll(FXCollections.observableArrayList(page));
+        return;
+      }
 
-              clanMenu.getItems().addAll(FXCollections.observableArrayList(toLeader, page));
+      MenuItem toLeader = new MenuItem(i18n.get("clan.toLeader"));
+      toLeader.setOnAction(event -> onClanTagClicked());
 
+      clanMenu.getItems().addAll(FXCollections.observableArrayList(toLeader, page));
 
-        }
 
       };
       executorService.submit(setMenuItems);
```

Comment:
Instead of using an executorService to run these, clanService should return a `CompletableFuture<Clan>` on which you can invoke `.thenAccept(clan -> { ... your code ... });`